src/cls/rbd/cls_rbd.cc

```diff
@@ -2675,6 +2693,51 @@ int metadata_get(cls_method_context_t hctx, bufferlist *in, bufferlist *out)
   return 0;
 }
 
+int get_snapshot_limit(cls_method_context_t hctx, bufferlist *in,
+		       bufferlist *out)
+{
+  int rc;
+  uint64_t snap_limit;
+
+  rc = read_key(hctx, "snap_limit", &snap_limit);
+  if (rc == -ENOENT) {
+    rc = 0;
+    ::encode(UINT64_MAX, *out);
+  } else {
```

Comment:
```snap_limit``` would be uninitialized in that case.  I think it would be good to return the error code if "rc < 0 && rc != -ENOENT" since it is really pointing to a failure in the OSD.